src/promise-katas-2.js

```diff
@@ -25,17 +25,31 @@ fetch("joke", "question").then(<insert your callback function>)
 // 1 Create a function that uses the fetch function to make a request to the "food" URL and returns
 // the data - expected return value "Cheese" of type String
 
-const food = () => {};
+const food = () => {
+const foodData = fetch("food").then((item) => {return item.data});
+return foodData;
+};
+
 
 // 2 Create a function that uses the fetch function to make a request to the "cats" URL and returns
 // a list of cats in alphabetical order - expected return value ["Bandit", "Berry", "Puss in boots", "Smokey"] of type Array
 
-const cat = () => {};
+const cat = () => {
+  const catList = fetch("cats").then((listOfCats) => {return listOfCats.data.cats.sort()});
+  return catList;
+};
 
 // 3 Create a function that uses the fetch function to make a request to the "dogs" URL and returns
 // the naughtiest dog - expected return value {name: "Mutley", naughty: 10} of type Object
 
-const dog = () => {};
+const dog = () => {
+  function compareNaughtiness(a, b) {return b.naughty - a.naughty}
+  const naughtyPooch = fetch("dogs")
+  .then((listOfDogs) => {return listOfDogs.data.dogs})
+  .then((dogList) => {return dogList.sort((compareNaughtiness))})
+  .then((naughtiest) => {return naughtiest[0]})
+  return naughtyPooch;
```

Comment:
Your solution works perfectly but it could be refactored by using the reduce method, or the find method.
You could fetch dogs, then you could use the find method on the response, or reduce on the response: